index.js

```diff
@@ -339,20 +356,21 @@ function flameGraph (opts) {
           data
             .sum(function (d) {
               // If this is the ancestor of a focusedFrame frame, use the same value (width) as the focusedFrame frame.
-              if (d.fade) return d.children.reduce(sumChildValues, 0)
+              if (d.fade) return 0
+              // When collapsing hidden nodes, they don't have a width; d3 will sum up their children's widths
+              if (d.hide && collapseHiddenNodeWidths) return 0
 
               // d3 sums value + all child values to get the value for a node,
               // we can set `value = specifiedValue - all child values` to counteract that.
               // the `.value`s in our data already include the sum of all child values.
-              const childValues = d.children
-                ? d.children.reduce(sumChildValues, 0)
-                : 0
-              return d.value - childValues
+              return d.value - sumChildValues(d)
             })
             .sort(doSort)
 
           // Make "all stacks" as wide as every visible stack.
-          data.value = data.children ? data.children.reduce(sumChildValues, 0) : 0
+          data.value = data.children
+            ? data.children.reduce((acc, node) => acc + node.value, 0)
```

Comment:
`sumChildValues` only accidentally worked here, here we're dealing with d3 node wrappers whereas sumChildValues expects 0x-style node data